docusaurus/docs/Android/03-ui/06-guides/09-adding-custom-attachments-v2.mdx

```diff
@@ -0,0 +1,454 @@
+
+In this guide, we'll demonstrate how to build a date sharing feature. Chat users will be able to pick a date from a calendar, preview their selection in the message input and see the sent attachment within a message in the message list.
+
+This involves the following steps:
+1. Customizing the message composer so that it is capable of sending messages with `date` attachments.
+2. Adding support for `date` attachments in the message list.
+
+:::note
+In this guide, we'll show only the main points concerning custom attachments. Smaller parts will be omitted for the sake of being concise.
+
+You can find the full code from this guide on [GitHub](https://github.com/GetStream/Android-Samples/tree/main/custom-attachments-v2).
+:::
+
+## Sending Date Attachments
+
+First of all, you'll need to customize the leading content of [MessageComposerView](../04-message-components/05-message-composer.mdx) to let the user pick a date using the date picker button. We'll take the [DefaultMessageComposerLeadingContent](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/composer/content/DefaultMessageComposerLeadingContent.kt) as a reference and add a new date picker button:
+
+```xml
+<merge xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <androidx.appcompat.widget.LinearLayoutCompat
+        android:layout_width="wrap_content"
+        android:layout_height="match_parent"
+        android:gravity="bottom"
+        android:orientation="horizontal"
+        android:paddingHorizontal="4dp"
+        android:paddingVertical="12dp">
+
+        <!-- The new date picker button -->
+        <ImageView
+            android:id="@+id/calendarButton"
+            android:layout_width="32dp"
+            android:layout_height="32dp"
+            android:padding="4dp"
+            android:src="@drawable/ic_calendar"
+            tools:ignore="ContentDescription" />
+
+        <ImageView
+            android:id="@+id/attachmentsButton"
+            android:layout_width="32dp"
+            android:layout_height="32dp"
+            android:padding="4dp"
+            android:src="@drawable/stream_ui_ic_attach"
+            tools:ignore="ContentDescription" />
+
+        <ImageView
+            android:id="@+id/commandsButton"
+            android:layout_width="32dp"
+            android:layout_height="32dp"
+            android:padding="4dp"
+            android:src="@drawable/stream_ui_ic_command"
+            tools:ignore="ContentDescription" />
+
+    </androidx.appcompat.widget.LinearLayoutCompat>
+
+</merge>
+```
+
+```kotlin
+class CustomMessageComposerLeadingContent : FrameLayout, MessageComposerContent {
+
+    private lateinit var binding: CustomMessageComposerLeadingContentBinding
```

Comment:
Maybe we should add the name for this  file and make a note that we are using viewbinding.

This probably won't throw off more experiences users, but I'm worried less experienced might for some reason presume `CustomMessageComposerLeadingContentBinding` is just a part of our SDK or something.

This is not a necessity but mostly a safeguard against confusing less experienced devs.